src/point_one/fusion_engine/messages/configuration.h

```diff
@@ -391,6 +391,356 @@ struct alignas(4) CoarseOrientation {
+ */
+struct alignas(4) SetOutputInterfaceStreamsMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE =
+      MessageType::SET_OUTPUT_INFERFACE_STREAMS;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+  /**
+   * The type of action this configuration message applies to the
+   * previous list of streams.
+   */
+  UpdateAction update_action = UpdateAction::REPLACE;
+  uint8_t reserved[3] = {0};
+
+  /**
+   * The configuration for an output interface to apply.
+   */
+  OutputInterfaceConfig output_interface_data;
+};
+
+/**
+ * @brief Query the sets of output streams enabled each output
+ *        interface (@ref
+ *        MessageType::GET_OUTPUT_INFERFACE_STREAMS, version 1.0).
+ * @ingroup config_and_ctrl_messages
+ *
+ * The device will respond with a @ref OutputInterfaceStreamsDataMessage
+ * containing the values.
+ */
+struct alignas(4) GetOutputInterfaceStreamsMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE =
+      MessageType::GET_OUTPUT_INFERFACE_STREAMS;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+
+  /** The config source to request data from (active, saved, etc.). */
+  ConfigurationSource request_source = ConfigurationSource::ACTIVE;
+
+  uint8_t reserved[3] = {0};
+};
+
+/**
+ * @brief Response to a @ref GetOutputInterfaceStreamsMessage request (@ref
+ *        MessageType::OUTPUT_INFERFACE_STREAMS_DATA, version 1.0).
+ * @ingroup config_and_ctrl_messages
+ *
+ * This message is followed by `N` @ref OutputInterfaceConfig objects, where
+ * `N` is equal to @ref number_of_interfaces. Each of these interfaces is
+ * variable size, and the sum of the objects should add up to the message size
+ * from the header.
+ *
+ * For example if the @ref number_of_interfaces is 2 and both interfaces have
+ * two streams the payload will look as follows:
+ *
+ * ```
+ * {MessageHeader, OutputInterfaceStreamsDataMessage, OutputInterfaceConfig,
+ *  uint8_t, uint8_t, OutputInterfaceConfig, uint8_t, uint8_t}
+ * ```
+ */
+struct alignas(4) OutputInterfaceStreamsDataMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE = MessageType::OUTPUT_INFERFACE_STREAMS_DATA;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+
+  /** The source of the parameter value (active, saved, etc.). */
+  ConfigurationSource config_source = ConfigurationSource::ACTIVE;
+
+  /**
+   * Set to `true` if the active configuration differs from the saved
+   * configuration for this parameter.
+   */
+  bool active_differs_from_saved = false;
+
+  /** The size of the parameter value, output_interface_data (in bytes). */
+  uint8_t number_of_interfaces = 0;
+
+  uint8_t reserved[1] = {0};
+
+  /**
+   * A pointer to the beginning of the interface data.
+   */
+  // Note: This causes a compiler error on MSVC so it is not included:
+  //       https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2233
+  // OutputInterfaceConfig output_interface_data[0];
```

Comment:
I resolved this already in other classes -- see [commit `8221a65`](https://github.com/PointOneNav/fusion-engine-client/commit/8221a65e5cf816734061c1672d47a97e73f35ec4)

Reply:
This actually is slightly different. If you look at https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2233 this issue is that MSVC gives an unsurpressable error if you make an array of a type with a zero length array. I'm not sure why this is different, but I couldn't find any work around for adding a MSVC compatible way to do this.